src/librustdoc/html/static/main.js

```diff
@@ -890,6 +890,22 @@
         return "\u2212"; // "\u2212" is '−' minus sign
     }
 
+    $(".search-input").on("keyup",function() {
+        if ($(this).val().length === 0) {
+            window.history.pushState("", "std - Rust", "?search=");
```

Comment:
Either use `replaceState`, or ensure that we only push at most 1 instance of this. Otherwise we just mess up the browser’s history stack.